x/ccv/provider/keeper/throttle.go

```diff
@@ -98,22 +98,23 @@ func (k Keeper) HandlePacketDataForChain(ctx sdktypes.Context, consumerChainID s
 }
 
 // InitializeSlashMeter initializes the slash meter to it's max value (also its allowance),
-// and sets the last replenish time to the current block time.
+// and sets the next replenish time one replenish period from block time.
 func (k Keeper) InitializeSlashMeter(ctx sdktypes.Context) {
 	k.SetSlashMeter(ctx, k.GetSlashMeterAllowance(ctx))
-	k.SetLastSlashMeterFullTime(ctx, ctx.BlockTime())
+	k.SetSlashMeterReplenishTimeCandidate(ctx, ctx.BlockTime().Add(k.GetSlashMeterReplenishPeriod(ctx)))
 }
 
 // CheckForSlashMeterReplenishment checks if the slash meter should be replenished, and if so, replenishes it.
-// Note: initial "last slash meter full time" is set in InitGenesis.
+// Note: initial slash meter replenish candidate time is set in InitGenesis.
 func (k Keeper) CheckForSlashMeterReplenishment(ctx sdktypes.Context) {
 
-	lastFullTime := k.GetLastSlashMeterFullTime(ctx)
 	replenishPeriod := k.GetSlashMeterReplenishPeriod(ctx)
 
-	// Replenish slash meter if enough time has passed since the last time it was full.
-	if ctx.BlockTime().UTC().After(lastFullTime.Add(replenishPeriod)) {
+	// Replenish slash meter if current time is equal to or after the current replenish candidate time.
+	if !ctx.BlockTime().UTC().Before(k.GetSlashMeterReplenishTimeCandidate(ctx)) {
```

Comment:
Nit:
`After()` is easier to parse than `!Before()`

Reply:
Agreed, but I need logic similar to "after or equal to" here

We could use an OR statement, either of [these options](https://stackoverflow.com/questions/69818655/how-to-do-greater-than-or-equal-operation-on-time-objects-in-go) work